### DISCUSSION TITLE
How do I change the path when transmuxing from rtmp to hls

### DISCUSSION THREAD

**Discussion by Tagvi** [bluenviron/mediamtx#1770] [Questions]
When I stream from OBS with these settings:
Server: rtmp://localhost:1935/username
Stream Key: xyz
The authentication URL gets hit with the following path: username/xyz and once it sends a 200 response code, the stream becomes accessible from http://localhost:8888/username/xyz/index.m3u8.
I want to access the stream without showing the stream key, which means instead of the path to get the HLS stream being username/xyz/index.m3u8, I want it to be username/index.m3u8
One option is using my own FFMPEG script, but I wonder if there is a better way

**Comment by aler9** [comment]
Hello, in my opinion you can setup authentication instead of using the stream key, which in case of this server server just adds an additional element to the stream path.

Server configuration:

```yml
paths:
  username:
    publishUser: myuser
    publishPass: mypass
```

OBS configuration:

```
Server: rtmp://localhost:1935/username?user=myuser&pass=mypass
Stream Key: 
```

Result:

```
http://localhost:8888/username/index.m3u8
```

[a comment by an automated account is not shown]
